### ISSUE TITLE
Sidekiq service names need to be explicitly set

### ISSUE THREAD

**Issue by konto-andrzeja** [DataDog/dd-trace-rb#2518]
I have a quite unusual bug report, because I found a fix and maybe it's documented somewhere, but I didn't find it, so I just thought I'll share the details. Maybe updating the docs will be enough to make the proper solution easier to find.

**Current behaviour**
When I set up Sidekiq tracing as described [in the docs](https://github.com/DataDog/dd-trace-rb/blob/master/docs/GettingStarted.md#sidekiq), it doesn't work. Neither `sidekiq` nor `sidekiq-client` services are visible in Datadog. That's because the respective service names are `nil`, even though according to [upgrade docs](https://github.com/DataDog/dd-trace-rb/blob/master/docs/UpgradeGuide.md#service-naming) application service name should be used as a fallback.

**Expected behaviour**
Sidekiq services should be visible in Datadog.

**Steps to reproduce**
Set up Sidekiq like described in the docs:
```ruby
require 'ddtrace'

Datadog.configure do |c|
  c.tracing.instrument :sidekiq
end
```

**Steps to fix**
Add additional params to set service names explicitly:
```ruby
require 'ddtrace'

Datadog.configure do |c|
  c.tracing.instrument :sidekiq, service_name: 'sidekiq', client_service_name: 'sidekiq-client'
end
```

**Environment**

* **ddtrace version:** 1.8.0
* **Ruby version:** 3.0.2

**Comment by marcotc** [comment]
👋 @konto-andrzeja, thanks for this report.

> that's because the respective service names are `nil`

How did you figure out that the service names were set to the value `nil`? Where did you intercept the code execution to collect this value?

**Comment by konto-andrzeja** [comment]
> Where did you intercept the code execution to collect this value?

https://github.com/DataDog/dd-trace-rb/blob/master/lib/datadog/tracing/contrib/sidekiq/client_tracer.rb#L17
and
https://github.com/DataDog/dd-trace-rb/blob/master/lib/datadog/tracing/contrib/sidekiq/server_tracer.rb#L20

`@sidekiq_service` is `nil` in both cases if the service names are not set explicitly.

**Comment by TonyCTHsu** [comment]
I think we should have default value for those service names

https://github.com/DataDog/dd-trace-rb/blob/master/lib/datadog/tracing/contrib/sidekiq/configuration/settings.rb#L35-L36

**Comment by konto-andrzeja** [comment]
> Could you put a break point at [here](https://github.com/DataDog/dd-trace-rb/blob/master/lib/datadog/core/configuration/components.rb#L80) and verified the return value for `settings.service` ?

When my config looks like this:
```ruby
require 'ddtrace'

Datadog.configure do |c|
  c.tracing.instrument :sidekiq
end
```
Then `setting.service` is the process name, as described in the docs.

When the config looks like this:
```ruby
require 'ddtrace'

Datadog.configure do |c|
  c.service = 'MyServiceName'
  c.tracing.instrument :sidekiq
end
```
Then `setting.service` is `MyServiceName`.

But when I tested those configs, there were no `sidekiq`, `sidekiq-client`, nor `MyServiceName` services present in Datadog. I'm not 100% sure about the last one, but I'm pretty sure I checked that scenario. I won't be able to recheck, since I already deployed the fixed version and I don't want to break it again.

**Comment by marcotc** [comment]
Hey @konto-andrzeja, the Sidekiq spans are being produced so it's a matter of figuring out what "service" is assigned to them.

Can you please take a look at your Trace Search page, and see if you can spot any Sidekiq spans, like so: https://app.datadoghq.com/apm/traces?query=operation_name%3Asidekiq.job&spanType=all

The `Service` column will tell you which service the Sidekiq trace is assigned to ("sidekiq" is the service name in my example below):
![Screen Shot 2023-01-05 at 11 43 05 AM](https://user-images.githubusercontent.com/583503/210866493-49041966-b640-465a-8d39-b9b9e58eece8.png)

Take a look at your Trace Search page and let us know what service name you have configured.

**Comment by konto-andrzeja** [comment]
@marcotc You are right! I found Sidekiq jobs under the default service, named after the process. It's not really convenient to have everything stuffed under one service, so I'll stick with the explicit names and separate services, but it's great to understand what was going on. Thanks!

While we're on this topic, do you know why should `c.service` be set explicitly? It's recommended [here](https://github.com/DataDog/dd-trace-rb/blob/master/docs/UpgradeGuide.md#service-naming). I'd rather have a separate service for each process.

**Comment by marcotc** [comment]
> While we're on this topic, do you know why should `c.service` be set explicitly? It's recommended [here](https://github.com/DataDog/dd-trace-rb/blob/master/docs/UpgradeGuide.md?rgh-link-date=2023-01-06T15%3A17%3A17Z#service-naming). I'd rather have a separate service for each process.

@konto-andrzeja, ideally your service is one "service" entity in Datadog, and it has different operations like so: 
![Screen Shot 2023-01-09 at 2 02 01 PM](https://user-images.githubusercontent.com/583503/211417400-fcfbcdd7-8b87-4a39-8463-157ccddb493f.png)

It's common to have different service names for different application entry-points (`my-service-sidekiq` for a Sidekiq runner process).

At the end of day, whatever works best for your application is what you should use :)
